plugins/fr.obeo.dsl.designer.sample.flow.design/description/flow.odesign

```diff
@@ -608,6 +608,25 @@
               </firstModelOperations>
             </initialOperation>
           </ownedTools>
+          <ownedTools xsi:type="tool:NodeCreationDescription" name="ProcessorWithConnection" label="Processor with Connection" nodeMappings="//@ownedViewpoints[name='Exchanges']/@ownedRepresentations[name='Topography_auto_layout']/@defaultLayer/@containerMappings[name='System']/@subNodeMappings[name='Processor']">
+            <variable name="container">
+              <subVariables xsi:type="tool_1:SelectModelElementVariable" name="selectedProcessor" candidatesExpression="aql:self.getRootElement().eAllContents(flow::Processor)" message="A new Processor will be created. Please select an existing Processor to connect to it:"/>
```

Comment:
I think AQL's, `eContainer(flow::System)` would work too and avoid a custom service.

Reply:
I don't think so, because flow::CompositeProcessor are also flow::Systems and flow::Systems can contain flow::Systems.